src/script/interpreter.cpp

```diff
@@ -1432,6 +1446,8 @@ bool ScriptMachine::Step()
 
                     if ((flags & SCRIPT_ENABLE_SCHNORR_MULTISIG) && stacktop(-idxDummy).size() != 0)
                     {
+                        if (!allNull)
+                            stats.consensusSigOpCount += nSigsCount; // 2020-05-15 sigchecks consensus rule
```

Comment:
This check is useless for schnorr multisig. Having all signatures equal to null is only possible if also the null dummy element is null, but [according to the spec](https://github.com/bitcoincashorg/bitcoincash.org/blob/master/spec/2019-11-15-schnorrmultisig.md#specification) the with schnorr musig the dummy element cannot be null, otherwise the tx would invalid. 

This is the case also in BU, see the [end of this if branch](https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/2122/files#diff-be2905e2f5218ecdbe4e55637dac75f3L1496)

```c++  
                            // Check signature
                            if (!checker.CheckSig(vchSig, vchPubKey, scriptCode))
                            {
                                // This can fail if the signature is empty, which also is a NULLFAIL error as the
                                // bitfield should have been null in this situation.
                                return set_error(serror, SCRIPT_ERR_SIG_NULLFAIL);
                            }
```

Hence I guess we could move `allNull` boolean computation inside the ECDSA if statement branch starting at line 1510. So that we compute only in case of ECDSA signatures check.

Reply:
true, good analysis.  But I need to use allNull again below.  I refactored to avoid that.